controllers/chat/moveToAnon.js

```diff
@@ -54,102 +65,93 @@ const moveToAnon = async (req, res) => {
+        checkChatAnonUserInfo?.anon_user_info_color_name;
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code =
+        checkChatAnonUserInfo?.anon_user_info_emoji_code;
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name =
+        checkChatAnonUserInfo?.anon_user_info_emoji_name;
+
+      newStateMemberWithAnonInfo[req.userId].user.name =
+        'Anonymous ' + checkChatAnonUserInfo?.anon_user_info_emoji_name;
+      newStateMemberWithAnonInfo[req.userId].user.username =
+        'Anonymous ' + checkChatAnonUserInfo?.anon_user_info_emoji_name;
+
+      await ChatAnonUserInfo.create({
+        channel_id: newChannel.id,
+        my_anon_user_id: req.userId,
+        target_user_id: prevTargetUser.user_id,
+        anon_user_info_color_code: checkChatAnonUserInfo?.anon_user_info_color_code,
+        anon_user_info_color_name: checkChatAnonUserInfo?.anon_user_info_color_name,
+        anon_user_info_emoji_code: checkChatAnonUserInfo?.anon_user_info_emoji_code,
+        anon_user_info_emoji_name: checkChatAnonUserInfo?.anon_user_info_emoji_name
+      });
+    } else {
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code = anon_user_info_color_code;
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_color_name = anon_user_info_color_name;
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code = anon_user_info_emoji_code;
+      newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name = anon_user_info_emoji_name;
+
+      newStateMemberWithAnonInfo[req.userId].user.name = 'Anonymous ' + anon_user_info_emoji_name;
+      newStateMemberWithAnonInfo[req.userId].user.username =
+        'Anonymous ' + anon_user_info_emoji_name;
+
+      await ChatAnonUserInfo.create({
+        channel_id: newChannel.id,
+        my_anon_user_id: req.userId,
+        target_user_id: prevTargetUser.user_id,
+        anon_user_info_color_code: anon_user_info_color_code,
+        anon_user_info_color_name: anon_user_info_color_name,
+        anon_user_info_emoji_code: anon_user_info_emoji_code,
+        anon_user_info_emoji_name: anon_user_info_emoji_name
+      });
+    }
 
     try {
+      let anonDetailInfo = {
+        anon_user_info_color_code: newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code,
+        anon_user_info_color_name: newStateMemberWithAnonInfo[req.userId].anon_user_info_color_name,
+        anon_user_info_emoji_code: newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code,
+        anon_user_info_emoji_name: newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name
+      };
+
       await newChannel.updatePartial({
         set: {
           channel_type: CHANNEL_TYPE.ANONYMOUS,
-          anon_user_info_color_code,
-          anon_user_info_color_name,
-          anon_user_info_emoji_code,
-          anon_user_info_emoji_name,
+          ...anonDetailInfo,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code is creating an `anonDetailInfo` object and updating the channel with this new information. It would be good to add some error handling here in case these operations fail.
```diff
+ try {
     let anonDetailInfo = {
       anon_user_info_color_code: newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code,
       anon_user_info_color_name: newStateMemberWithAnonInfo[req.userId].anon_user_info_color_name,
       anon_user_info_emoji_code: newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code,
       anon_user_info_emoji_name: newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name
     };

     await newChannel.updatePartial({
       set: {
         channel_type: CHANNEL_TYPE.ANONYMOUS,
        ...anonDetailInfo,
        better_channel_member: newStateMemberWithAnonInfo
      }
    });
+ } catch (error) {
+   console.error('Error updating channel:', error);
+   throw error;
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->